optimize/upgrade/src/test/java/io/camunda/optimize/upgrade/AbstractUpgradeIT.java

```diff
@@ -0,0 +1,354 @@
+ import io.camunda.optimize.upgrade.service.UpgradeValidationService;
+ import io.camunda.optimize.upgrade.steps.UpgradeStep;
+ import io.camunda.optimize.upgrade.steps.document.DeleteDataStep;
+ import io.camunda.optimize.upgrade.steps.document.InsertDataStep;
+ import io.camunda.optimize.upgrade.steps.document.UpdateDataStep;
+ import io.camunda.optimize.upgrade.steps.schema.DeleteIndexIfExistsStep;
+ import io.camunda.optimize.upgrade.util.UpgradeUtil;
+ import java.io.IOException;
+ import java.util.Arrays;
+ import java.util.Collections;
+ import java.util.List;
+ import java.util.Optional;
+ import lombok.SneakyThrows;
+ import org.apache.http.HttpEntity;
+ import org.apache.http.client.methods.HttpPost;
+ import org.apache.http.entity.ContentType;
+ import org.apache.http.nio.entity.NStringEntity;
+ import org.elasticsearch.ElasticsearchStatusException;
+ import org.elasticsearch.action.admin.indices.alias.Alias;
+ import org.elasticsearch.action.admin.indices.refresh.RefreshRequest;
+ import org.elasticsearch.action.bulk.BulkRequest;
+ import org.elasticsearch.action.get.GetRequest;
+ import org.elasticsearch.action.get.GetResponse;
+ import org.elasticsearch.action.index.IndexRequest;
+ import org.elasticsearch.action.search.SearchRequest;
+ import org.elasticsearch.action.search.SearchResponse;
+ import org.elasticsearch.client.Request;
+ import org.elasticsearch.client.indices.CreateIndexRequest;
+ import org.elasticsearch.client.indices.GetIndexRequest;
+ import org.elasticsearch.client.indices.GetIndexResponse;
+ import org.elasticsearch.common.settings.Settings;
+ import org.elasticsearch.index.query.QueryBuilders;
+ import org.elasticsearch.index.reindex.DeleteByQueryRequest;
+ import org.elasticsearch.search.SearchHit;
+ import org.elasticsearch.search.builder.SearchSourceBuilder;
+ import org.elasticsearch.xcontent.XContentBuilder;
+ import org.elasticsearch.xcontent.XContentType;
+ import org.junit.jupiter.api.AfterEach;
+ import org.junit.jupiter.api.BeforeEach;
+ import org.mockserver.integration.ClientAndServer;
+ import org.mockserver.model.HttpRequest;
+
+ public abstract class AbstractUpgradeIT {
+
+   protected static final MetadataIndexES METADATA_INDEX = new MetadataIndexES();
+
+   protected static final String FROM_VERSION = "2.6.0";
+   protected static final String INTERMEDIATE_VERSION = "2.6.1";
+   protected static final String TO_VERSION = "2.7.0";
+
+   protected static final IndexMappingCreator TEST_INDEX_V1 = new UserTestIndex(1);
+   protected static final IndexMappingCreator TEST_INDEX_V2 = new UserTestIndex(2);
+   protected static final IndexMappingCreator TEST_INDEX_WITH_UPDATED_MAPPING_V2 =
+       new UserTestUpdatedMappingIndex();
+   protected static final IndexMappingCreator TEST_INDEX_WITH_TEMPLATE_V1 =
+       new UserTestWithTemplateIndex();
+   protected static final IndexMappingCreator TEST_INDEX_WITH_TEMPLATE_UPDATED_MAPPING_V2 =
```

Comment:
many of these are unused. Do you need them?

Reply:
I didnt add this, I temporarily uncommented the file to run the migrationIT and then commented it out again, but looks like now it has both versions of the file. Will remove this one, I can review if it can be refactored

Reply:
Ah, so theyre unsued because the other IT that are using them are commented out. Leaving as is and just commenting out the file again then